templates/VPC/vpc-20-private.yaml

```diff
@@ -0,0 +1,396 @@
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  Creates a VPC with public and private subnets and a managed NAT gateway.
+Parameters:
+  VpcName:
+    Description: The VPC name (i.e. Synapse-Prod)
+    Type: String
+  VpcSubnetPrefix:
+    Description: The VPC subnet prefix (i.e. 10.255)
+    Type: String
+  PublicSubnetZones:
+    Description: Availability zones for public subnets
+    Type: List<AWS::EC2::AvailabilityZone::Name>
+    ConstraintDescription: List of Availability Zones in a region, such as us-east-1a, us-east-1b, us-east-1c
+    Default: "us-east-1a, us-east-1b, us-east-1c"
+  PrivateSubnetZones:
+    Description: Availability zones for private subnets
+    Type: List<AWS::EC2::AvailabilityZone::Name>
+    ConstraintDescription: List of Availability Zones in a region, such as us-east-1a, us-east-1b, us-east-1c
+    Default: "us-east-1a, us-east-1b, us-east-1c"
+  VpnCidr:
+    Description: CIDR of the (sophos-utm) VPN
+    Type: String
+    Default: "10.1.0.0/16"
+    AllowedPattern: ^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/(1[6-9]|2[0-8]))$
+    ConstraintDescription: CIDR block parameter must be in the form x.x.x.x/16-28
+Mappings:
+  SubnetConfig:
+    VPC:
+      CIDR: "0.0/20"  # 10.255.0.0 - 10.255.15.255
+    Private:
+      CIDR: "0.0/22"  # 10.255.0.0 - 10.255.3.255
```

Comment:
my preference is to split a /20 CIDR into 4 equal subnets, https://pasteboard.co/JZ8SNxK.png

Reply:
@zaro0508, I agree that would be tidier and preferable, but according to the AWS docs a public subnet is necessary for a NAT gateway. @ahayden suggested we go with 4096 addresses (`/20`), and I split them in this way to not waste too many on the public subnet. Alternately, perhaps I should go to  `/19`?

Reply:
sounds good, however there might be a one off error in your comment of ip ranges? https://pasteboard.co/JZ9vjzB.png

Reply:
don't you want to doc the usable IP range instead of the full range?  https://www.davidc.net/sites/default/subnets/subnets.html?network=192.168.0.0&mask=20&division=11.351

Reply:
Sure, I can make that change